Gems/Atom/RPI/Code/Source/RPI.Public/RPIUtils.cpp

```diff
@@ -606,24 +607,23 @@ namespace AZ
                 return false;
             }
 
-            const AZ::RHI::ImageDescriptor& imageDescriptor = imageAsset->GetImageDescriptor();
+            auto imageDescriptor = imageAsset->GetImageDescriptorForMipLevel(mip);
 
-            size_t outValuesIndex = 0;
             for (uint32_t y = topLeft.second; y < bottomRight.second; ++y)
             {
                 for (uint32_t x = topLeft.first; x < bottomRight.first; ++x)
                 {
                     size_t imageDataIndex = Internal::GetImageDataIndex(imageDescriptor, x, y, componentIndex);
 
-                    auto& outValue = outValues[outValuesIndex++];
-                    outValue = Internal::RetrieveFloatValue(imageData.data(), imageDataIndex, imageDescriptor.m_format);
+                    float value = Internal::RetrieveFloatValue(imageData.data(), imageDataIndex, imageDescriptor.m_format);
```

Comment:
Just a reminder that this should be refactored at some point - you shouldn't need to calculate this for each pixel. All you really need is a starting address and a stride.

Reply:
I actually requested that @cgalvan leave it this way, because with my upcoming PR that adds BC1 support, it's _not_ just a starting address and stride any more. Compressed blocks need a bit more calculation.